doc/whats-new.rst

```diff
@@ -53,6 +53,10 @@ Bug fixes
 - Fix applying function with non-xarray arguments using :py:func:`xr.map_blocks`.
   By `Cindy Chiao <https://github.com/tcchiao>`_.
 
+- `dt.season <https://xarray.pydata.org/en/stable/generated/xarray.DataArray.dt.season.html>`_  can now handle NaN and NaT.  (:pull:`5876`).
+  By `Pierre Loicq <https://github.com/pierreloicq>`_ and `Spencer Clark <https://github.com/spencerkclark>`_.
```

Comment:
This is super minor, but I just want to keep things consistent.  While I appreciate the sentiment, we typically only credit reviewers in the what's new in exceptional circumstances.
```suggestion
  By `Pierre Loicq <https://github.com/pierreloicq>`_.
```